azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+package azurerm
+
+import (
+	"bytes"
+	"sort"
+	"strconv"
+	"time"
+
+	"fmt"
+
+	"net/http"
+
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/arm/insights"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmAutoscaleSetting() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutoscaleSettingCreateOrUpdate,
+		Read:   resourceArmAutoscaleSettingRead,
+		Update: resourceArmAutoscaleSettingCreateOrUpdate,
+		Delete: resourceArmAutoscaleSettingDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": locationSchema(),
+			"profile": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 20, // https://msdn.microsoft.com/en-us/library/azure/dn931928.aspx
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"capacity": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MinItems: 1,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"minimum": {
+										Type:     schema.TypeInt,
+										Required: true,
```

Comment:
Good catch. The value should be in the range of [1, 100] - yes, for VMSS you can go up to 1000 (but actually limited by the allowed core numbers of the subscription), but for Autoscale Settings, somehow it is limited to 100. On portal there is a input validation and the REST API returns error if the value exceeds 100.